openpype/hosts/houdini/plugins/create/create_unreal_staticmesh.py

```diff
@@ -0,0 +1,164 @@
+        """Add settings for users. """
+
+        attrs = super().get_pre_create_attr_defs()
+        createsubnetroot = BoolDef("createsubnetroot",
+                                   tooltip="Create an extra root for the "
+                                           "Export node when it's a "
+                                           "subnetwork. This causes the "
+                                           "exporting subnetwork node to be "
+                                           "represented in the FBX file.",
+                                   default=False,
+                                   label="Create Root for Subnet")
+        vcformat = EnumDef("vcformat",
+                           items={
+                               0: "Maya Compatible (MC)",
+                               1: "3DS MAX Compatible (PC2)"
+                           },
+                           default=0,
+                           label="Vertex Cache Format")
+        convert_units = BoolDef("convertunits",
+                                tooltip="When on, the FBX is converted"
+                                        "from the current Houdini "
+                                        "system units to the native "
+                                        "FBX unit of centimeters.",
+                                default=False,
+                                label="Convert Units")
+
+        return attrs + [createsubnetroot, vcformat, convert_units]
+
+    def get_dynamic_data(
+        self, variant, task_name, asset_doc, project_name, host_name, instance
+    ):
+        """
+        The default subset name templates for Unreal include {asset} and thus
+        we should pass that along as dynamic data.
+        """
+        dynamic_data = super(CreateUnrealStaticMesh, self).get_dynamic_data(
+            variant, task_name, asset_doc, project_name, host_name, instance
+        )
+        dynamic_data["asset"] = asset_doc["name"]
+        return dynamic_data
+
+    def get_parms(self, subset_name, pre_create_data):
+        """Get parameters values. """
+
+        # 1. get output path
+        output_path = hou.text.expandString(
+            "$HIP/pyblish/{}.fbx".format(subset_name))
+
+        # 2. get selection
+        selection = self.get_selection()
+
+        # 3. get Vertex Cache Format
+        vcformat = pre_create_data.get("vcformat")
+
+        # 4. get convert_units
+        convertunits = pre_create_data.get("convertunits")
+
+        # 5. get Valid Frame Range
+        trange = 1
+
+        # 6. get createsubnetroot
+        createsubnetroot = pre_create_data.get("createsubnetroot")
+
+        # parms dictionary
+        parms = {
+            "startnode": selection,
+            "sopoutput": output_path,
+            "vcformat": vcformat,
+            "convertunits": convertunits,
+            "trange": trange,
+            "createsubnetroot": createsubnetroot
+        }
+
+        return parms
```

Comment:
The `# get` commands plus the `get_parms` method name are confusing to me. It's not getting anything from an houdini parm but is just defining the default values. To me this would be just as readable if it were:
```python
        output_path = hou.text.expandString("$HIP/pyblish/{}.fbx".format(subset_name))
        instance_node.setParms({
            "startnode": selection,
            "sopoutput": output_path,
            "vcformat": pre_create_data.get("vcformat"),
            "convertunits": pre_create_data.get("convertunits"),
            "trange": 1,
            "createsubnetroot": pre_create_data.get("createsubnetroot")
        })
```

If there are specific parm names that are confusing, like `vcformat` I would likely comment like this:
```python
        output_path = hou.text.expandString("$HIP/pyblish/{}.fbx".format(subset_name))
        instance_node.setParms({
            "startnode": selection,
            "sopoutput": output_path,
            # vertex cache format
            "vcformat": pre_create_data.get("vcformat"),
            "convertunits": pre_create_data.get("convertunits"),
            # set render range to use frame range start-end frame
            "trange": 1,
            "createsubnetroot": pre_create_data.get("createsubnetroot")
        })
```

Reply:
man, I love your simple solutions!